### PR TITLE
GTK 4.14.1 and other updates

### PR DESCRIPTION
**Pull request by danyeaw**

### CODE REVIEW

**Comment by danyeaw** [comment]
We have run in to https://discourse.gnome.org/t/dealing-with-glib-and-gobject-introspection-circular-dependency/18701.

We'll have to figure out a way to build GLib twice, once without gi enabled, and a 2nd time with it enabled after gobject-introspection is built.

**Comment by nacho** [comment]
maybe having 2 projects? one called glib-base and the other glib?

**Review by nacho — APPROVED** [review]

**Comment by pbor** [comment]
I think we need a `glib-internal` or something similar package and we make gobject-introspection depend on it. Everything else keeps depending on glib. glib-internal is a copy of glib.

In the mean time I split out a PR with the updates that are unrelated

File: gvsbuild/projects/pango.py [hunk 1]
```diff
@@ -43,6 +44,8 @@ def __init__(self):
             enable_gi = "disabled"
 
         self.add_param(f"-Dintrospection={enable_gi}")
+        self.add_param("-Dfreetype=enabled")
+        self.add_param("-Dfontconfig=enabled")
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply gvsbuild/projects/pango.py]: Thanks pbor for the feedback, I dropped this commit.

**Comment by danyeaw** [comment]
I am still getting a build error during the 2nd build of GLib with introspection enabled, with an ImportError of giscanner.

I created an upstream bug report here:
https://gitlab.gnome.org/GNOME/glib/-/issues/3296

**Comment by danyeaw** [comment]
So far, we have fixes upstream for GLib and GTK. The next issue is here: https://gitlab.gnome.org/GNOME/gtksourceview/-/issues/360

**Comment by danyeaw** [comment]
This should now be good to go!

**Review by nacho — APPROVED** [review]
Great job @danyeaw !

**Comment by pbor** [comment]
While the patch here looks good to me, reading the tickets on gtk/pango, it sounds like gtk is still crashing on windows.

I think we should wait for that to be addressed before merging

**Comment by pbor** [comment]
https://gitlab.gnome.org/GNOME/pango/-/issues/789 actually just got merged, maybe we just need to pick that up

**Comment by danyeaw** [comment]
@pbor, I wasn't seeing that specific crash, but I agree that it is a good idea to pick up the patch in case it is impacting others. I added the patch to Pango. 👍

**Pull request merged**